meshroom/ui/qml/main.qml

```diff
@@ -475,15 +490,26 @@ ApplicationWindow {
 
     // Utility functions for elements in the menubar
 
-    function initFileDialogFolder(dialog) {
-        if(_reconstruction.graph && _reconstruction.graph.filepath) {
-            dialog.folder = Filepath.stringToUrl(Filepath.dirname(_reconstruction.graph.filepath));
+    function initFileDialogFolder(dialog, importImages = false) {
+        let folder = "";
+
+        if (_reconstruction.graph && _reconstruction.graph.filepath) {
+            folder = Filepath.stringToUrl(Filepath.dirname(_reconstruction.graph.filepath));
+            if (imagesFolder.toString() === "" && workspaceView.imageGallery.grid.itemAtIndex(0) !== null) {
+                imagesFolder = Filepath.stringToUrl(Filepath.dirname(workspaceView.imageGallery.grid.itemAtIndex(0).source));
```

Comment:
Why do we check the path of the first input image only if we have saved the current scene?

Reply:
In case `imagesFolder` was empty because the user had not imported any image yet and a project with images had been opened, the idea was to use the location of these images to set `imagesFolder` rather than waiting for the user to do a first "Import Images".

Waiting to have a _reconstruction.graph.filepath was initially done because we didn't want to use the location of dropped images as the base folder: it could only be "actively" set by using the "Import Images" action , or "passively" by opening a project file with images (and so dropping images in the gallery would never enter the `if (_reconstruction.graph.filepath)` condition). 

This makes less sense now that the locations are saved across sessions. Since the number of cases for which `imagesFolder` is not set is virtually 0 now (it will happen once the first time a user starts Meshroom, and never again unless the settings are reset, all 3 saved locations have become invalid or the user never uses the "Import Images" action), it might be better to move it outside of the `if`. That way we prevent as much as possible opening the dialog on Meshroom's install location for the rare cases in which `imagesFolder` is not set. 

The location will still be only saved across sessions if the "Import Images" dialog is used.